### PR TITLE
feat: health check for controller

### PR DESCRIPTION
**Pull request by zwForrest**
## Description
There are currently no health checks for crane-controller. This commit is to add health checks.

## Related Issues
#10

### CODE REVIEW

File: deploy/controller/deployment.yaml [hunk 1]
```diff
@@ -39,6 +39,19 @@ spec:
             requests:
               cpu: 200m
               memory: 200Mi
+          livenessProbe:
+            failureThreshold: 3
+            httpGet:
```
Comment (autumn0207) [inline deploy/controller/deployment.yaml:44]: Can you add them to controller's [helm charts](https://github.com/gocrane/helm-charts/blob/main/charts/scheduler/templates/controller-deployment.yaml) as well?

File: cmd/controller/app/server.go [hunk 2]
```diff
@@ -73,6 +75,14 @@ func Run(cc *config.CompletedConfig, stopCh <-chan struct{}) error {
 		panic(annotatorController.Run(int(cc.AnnotatorConfig.ConcurrentSyncs), stopCh))
 	}
 
+	healthMux := http.NewServeMux()
+	healthz.InstallHandler(healthMux, healthz.NamedCheck("crane-controller", healthz.PingHealthz.Check))
```
Comment (autumn0207) [inline cmd/controller/app/server.go:79]: Using `crane-controller` may semantically conflict with the controller in the main project, so it is better to use `scheduler-controller` or `crane-scheduler-controller` here

File: deploy/controller/deployment.yaml (hunk 1 above)
Comment (zwForrest) [reply deploy/controller/deployment.yaml:44]: ok

File: cmd/controller/app/server.go (hunk 2 above)
Comment (zwForrest) [reply cmd/controller/app/server.go:79]: ok

**Comment by autumn0207** [comment]
LGTM

**Pull request merged**